speechbrain/decoders/scorer.py

```diff
@@ -0,0 +1,850 @@
+# -*- coding: utf-8 -*-
+"""
+Token scorer abstraction and specifications.
+
+Authors:
+ * Adel Moumen 2022, 2023
+ * Sung-Lin Yeh 2021
+"""
+
+import torch
+import numpy as np
+import speechbrain as sb
+from speechbrain.decoders.ctc import CTCPrefixScore
+
+
+class BaseScorerInterface:
+    """A scorer abstraction to be inherited by other
```

Comment:
What is it usefull for? Add a description of what is the role of such a "scorer" what is the definition of SpeechBrain of a "scorer"?

Reply:
done